pkg/repo/mysql/message.go

```diff
@@ -0,0 +1,150 @@
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"github.com/google/uuid"
+	"go.expect.digital/translate/pkg/model"
+	"golang.org/x/text/language"
+)
+
+func (r *Repo) SaveMessages(ctx context.Context, serviceID uuid.UUID, messages *model.Messages) error {
+	_, err := r.LoadService(ctx, serviceID)
+	if err != nil {
+		return fmt.Errorf("repo: load service: %w", err)
+	}
+
+	tx, err := r.db.BeginTx(ctx, nil)
+	if err != nil {
+		return fmt.Errorf("repo: begin tx: %w", err)
+	}
+
+	defer tx.Rollback() //nolint:errcheck
+
+	// Check if message already exist
+	var messageID uuid.UUID
+
+	row := tx.QueryRowContext(
+		ctx,
+		`SELECT id FROM message WHERE service_id = UUID_TO_BIN(?) AND language = ?`,
+		serviceID.String(),
+		messages.Language.String(),
+	)
+
+	err = row.Scan(&messageID)
+
+	// If message does not exist, generate a new UUID for it
+	switch {
+	case errors.Is(err, sql.ErrNoRows):
+		messageID = uuid.New()
+	case err != nil:
+		return fmt.Errorf("repo: scan message: %w", err)
+	}
+
+	// Insert into message, ignore if already exists
```

Comment:
You know if it exists or it does not. Any network operation is expensive by default. Insert only if it does not exist!